src/components/selects/mixins/autocomplete.js

```diff
@@ -7,12 +13,8 @@ export default {
 
   methods: {
     filterSearch () {
-      return this.items.filter(i => {
-        const text = this.getText(i)
-        if (typeof text !== 'string') return false
-
-        return text.toLowerCase().indexOf(this.searchValue.toLowerCase()) !== -1
-      })
+      const query = this.searchValue.toLowerCase()
```

Comment:
imho query.toLowerCase() should be evaluated in the default filter function instead of filterSearch(), otherwise it's not possible to pass the case sensitive filter function